src/region/RegionContainer.tsx

```diff
@@ -31,7 +31,7 @@ export const mapStateToProps = (state: AppState, { page }: { page: number }): Pr
     activeAnnotationId: getActiveAnnotationId(state),
     annotations: getAnnotationsForLocation(state, page).filter(isRegion),
     isCreating: getAnnotationMode(state) === 'region',
-    staged: getCreatorStagedForLocation(state, page),
+    staged: getCreatorStaged(state),
```

Comment:
If we leave this as-is, I don't think we need some of the optional chaining conditions we're doing in `RegionAnnotations`. Consider reverting?

Reply:
It seems like we need a new action, `UPDATE_STAGED` that handles the merging of current/new data for the container. The container should only need to know about the staged item if it's actually rendering it.